### PR TITLE
RD-1609 Refactor exposing items from `Stage.Common`

### PR DESCRIPTION
**Pull request by kubama**
## Description
Refactored exposing items from Stage.Common

## Screenshots / Videos
N/A

## Checklist

- [X] My code follows the [UI Code Style](https://cloudifysource.atlassian.net/l/c/x8fq902p).
- [x] I verified that all tests and checks have passed.
- [X] I verified if that change requires any update in the [documentation](https://cloudifysource.atlassian.net/l/c/4XKtPocy).
- [X] I added proper labels to this PR.

## Tests
https://jenkins.cloudify.co/view/UI/job/Stage-UI-System-Test/2381/ - pass, the failing snapshot widget spec is unstable on master as well

 

## Documentation
N/A

### CODE REVIEW

File: test/cypress/support/filters.ts [hunk 1]
```diff
@@ -1,4 +1,5 @@
 import { addCommands, GetCypressChainableFromCommands } from 'cloudify-ui-common/cypress/support';
+import type { FilterRule } from 'widgets/common/src/filters/types';
 import { waitUntilEmpty } from './resource_commons';
 import type {} from '../../../widgets/common/src/filters';
```
Comment (qooban) [inline test/cypress/support/filters.ts:4]: Can you check if we still need this?

File: widgets/common/src/blueprints/BlueprintActions.ts [hunk 2]
```diff
@@ -1,5 +1,7 @@
 // eslint-disable-next-line max-classes-per-file
+import Consts from '../Consts';
 import DeploymentActions from '../deployments/DeploymentActions';
+import { Label } from '../labels/types';
```
Comment (qooban) [inline widgets/common/src/blueprints/BlueprintActions.ts:4]:
```suggestion
import type { Label } from '../labels/types';
```

File: widgets/common/src/deployments/DeploymentActions.ts [hunk 3]
```diff
@@ -1,5 +1,7 @@
 import type { Workflow } from '../executeWorkflow';
 import ExecutionActions from '../executions/ExecutionActions';
+import { Label } from '../labels/types';
+import { Site } from '../map/site';
```
Comment (qooban) [inline widgets/common/src/deployments/DeploymentActions.ts:4]:
```suggestion
import type { Label } from '../labels/types';
import type { Site } from '../map/site';
```

File: widgets/common/src/deploymentsView/detailsPane/drilldownButtons/SubdeploymentDrilldownButton.tsx [hunk 4]
```diff
@@ -1,6 +1,6 @@
 import type { FunctionComponent } from 'react';
 import { useMemo } from 'react';
-import { FilterRuleOperators, FilterRuleType } from '../../../filters/types';
+import { FilterRule, FilterRuleOperators, FilterRuleType } from '../../../filters/types';
```
Comment (qooban) [inline widgets/common/src/deploymentsView/detailsPane/drilldownButtons/SubdeploymentDrilldownButton.tsx:3]:
```suggestion
import type { FilterRule, FilterRuleOperators, FilterRuleType } from '../../../filters/types';
```

File: widgets/common/src/deploymentsView/index.ts [hunk 5]
```diff
@@ -1,33 +1,13 @@
 import * as DCommon from './common';
-import type { DeploymentsResponse } from './types';
-import { SharedDeploymentsViewWidgetConfiguration, sharedConfiguration, sharedDefinition } from './configuration';
-import './styles.scss';
+import { sharedConfiguration, sharedDefinition } from './configuration';
 import { DeploymentsView as DDeploymentsView } from './DeploymentsView';
+import './styles.scss';
 import { DeploymentStatus } from './types';
```
Comment (qooban) [inline widgets/common/src/deploymentsView/index.ts:5]:
```suggestion
import type { DeploymentStatus } from './types';
```

File: widgets/common/src/deploymentsView/map/DeploymentsMap.tsx [hunk 6]
```diff
@@ -1,6 +1,10 @@
 import { Dictionary, keyBy, partition } from 'lodash';
 import { FunctionComponent, useEffect, useMemo, useRef } from 'react';
 import type { Map } from 'react-leaflet';
+import DefaultTileLayer from '../../map/DefaultTileLayer';
+import { getMapOptions } from '../../map/options';
+import { isSiteWithPosition, Site, SiteWithPosition } from '../../map/site';
+import { invalidateSizeAfterDimensionsChange, WidgetDimensions } from '../../map/widget-dimensions';
```
Comment (qooban) [inline widgets/common/src/deploymentsView/map/DeploymentsMap.tsx:7]:
```suggestion
import { isSiteWithPosition } from '../../map/site';
import type { Site, SiteWithPosition } from '../../map/site';
import { invalidateSizeAfterDimensionsChange } from '../../map/widget-dimensions';
import type { WidgetDimensions } from '../../map/widget-dimensions';
```

File: widgets/common/src/deploymentsView/map/DeploymentsMap.tsx [hunk 7]
```diff
@@ -1,6 +1,10 @@
 import { Dictionary, keyBy, partition } from 'lodash';
 import { FunctionComponent, useEffect, useMemo, useRef } from 'react';
```
Comment (qooban) [inline widgets/common/src/deploymentsView/map/DeploymentsMap.tsx:2]:
```suggestion
import { keyBy, partition } from 'lodash';
import type { Dictionary } from 'lodash';
import { useEffect, useMemo, useRef } from 'react';
import type { FunctionComponent } from 'react';
```

File: widgets/common/src/deploymentsView/map/common.ts [hunk 8]
```diff
@@ -1,9 +1,10 @@
+import { SiteWithPosition } from '../../map/site';
```
Comment (qooban) [inline widgets/common/src/deploymentsView/map/common.ts:1]:
```suggestion
import type { SiteWithPosition } from '../../map/site';
```

File: widgets/common/src/deploymentsView/map/container.tsx [hunk 9]
```diff
@@ -3,6 +3,7 @@ import { useQuery } from 'react-query';
 import NoDataMessage from '../../components/NoDataMessage';
 import DeploymentActions from '../../deployments/DeploymentActions';
 import MapsActions from '../../map/MapsActions';
+import { Site } from '../../map/site';
```
Comment (qooban) [inline widgets/common/src/deploymentsView/map/container.tsx:6]:
```suggestion
import type { Site } from '../../map/site';
```

File: widgets/common/src/deploymentsView/map/marker/single-deployment-site.tsx [hunk 10]
```diff
@@ -1,4 +1,5 @@
 import { FunctionComponent, RefObject, useEffect, useRef } from 'react';
+import { siteToLatLng } from '../../../map/site';
 import { createMarkerIcon, MarkerIconColor } from '../../../map/MarkerIcon';
```
Comment (qooban) [inline widgets/common/src/deploymentsView/map/marker/single-deployment-site.tsx:3]:
```suggestion
import { useEffect, useRef } from 'react';
import type { FunctionComponent, RefObject } from 'react';
import { siteToLatLng } from '../../../map/site';
import { createMarkerIcon } from '../../../map/MarkerIcon';
import type { MarkerIconColor } from '../../../map/MarkerIcon';
```

File: widgets/common/src/filters/index.ts [hunk 11]
```diff
@@ -1,30 +1,14 @@
-import RulesForm from './RulesForm';
+import { filterIdQueryParameterName } from './common';
 import FilterActions from './FilterActions';
 import FilterIdDropdown from './FilterIdDropdown';
-import { FilterRule, FilterRuleOperators, FilterRuleType } from './types';
-import { filterIdQueryParameterName } from './common';
+import RulesForm from './RulesForm';
+import { FilterRuleOperators, FilterRuleType } from './types';
```
Comment (qooban) [inline widgets/common/src/filters/index.ts:5]:
```suggestion
import type { FilterRuleOperators, FilterRuleType } from './types';
```

File: widgets/serviceButton/src/ServiceButton.tsx [hunk 12]
```diff
@@ -1,12 +1,13 @@
 import type { SemanticCOLORS, SemanticICONS } from 'semantic-ui-react';
 import { FunctionComponent } from 'react';
```
Comment (qooban) [inline widgets/serviceButton/src/ServiceButton.tsx:2]:
```suggestion
import type { FunctionComponent } from 'react';
```

File: widgets/common/src/blueprintMarketplace/index.ts [hunk 13]
```diff
@@ -1,23 +1,7 @@
...
+export default {
     Modal,
     tabsConfig
 };
-
-declare global {
-    namespace Stage.Common {
-        namespace BlueprintMarketplace {
-            export type Tab = MarketplaceTab;
-            export type DisplayStyle = MarketplaceDisplayStyle;
```
Comment (qooban) [inline widgets/common/src/blueprintMarketplace/index.ts:14]: Good idea to stop exposing TS types through `Stage.Common` 👍🏻

**Review by qooban — COMMENTED** [review]
Well done 👍🏻 

Added - mostly minor - comments about `import` -> `import type` conversion which you can quickly apply using a batch suggestion commit. I hope that very soon we'll be able to automate that process with RD-2000.

File: test/cypress/support/filters.ts (hunk 1 above)
Comment (kubama) [reply test/cypress/support/filters.ts]: We don't, removed.

**Review by qooban — APPROVED** [review]
LGTM 👍🏻

**Review by jawamartus — APPROVED** [review]
Looks good!

**Pull request merged**